overlord/overlord_test.go

```diff
@@ -214,6 +214,35 @@ func (ovs *overlordSuite) TestEnsureLoopMediatedEnsureBeforeInEnsure(c *C) {
 	}
 }
 
+func (ovs *overlordSuite) TestEnsureLoopPrune(c *C) {
+	restoreIntv := overlord.MockPruneInterval(10*time.Millisecond, 5*time.Millisecond, 5*time.Millisecond)
+	defer restoreIntv()
+	o, err := overlord.New()
+	c.Assert(err, IsNil)
+
+	st := o.State()
+	st.Lock()
+	t1 := st.NewTask("foo", "...")
+	chg1 := st.NewChange("abort", "...")
+	chg1.AddTask(t1)
+	chg2 := st.NewChange("prune", "...")
+	chg2.SetStatus(state.DoneStatus)
+	st.Unlock()
+
+	o.Loop()
+	time.Sleep(50 * time.Millisecond)
```

Comment:
this kind of sleep based tests tend to be fragile on slow ci , not a blocker but we will have to revisit this (we have already a flaky one somewhere else under overlord that I need to take a look at)